system/include/emscripten/em_asm.h

```diff
@@ -7,6 +7,9 @@
 
 #pragma once
 
+// EM_ASM does not work strict C mode.
+#if defined(__cplusplus) || !defined(__STRICT_ANSI__)
```

Comment:
Can we invert the conditional here?
`#if !defined(__cplusplus) && defined(__STRICT_ANSI__)`

That moves the error definitions much closer to the comment...